backend/pkg/connect/get_connectors.go

```diff
@@ -222,32 +258,141 @@ func listConnectorsExpandedToClusterConnectorInfo(l map[string]con.ListConnector
 	connectorInfo := make([]ClusterConnectorInfo, 0, len(l))
 	for _, c := range l {
 		tasks := make([]ClusterConnectorTaskInfo, len(c.Status.Tasks))
+		connectorTaskErrors := make([]ClusterConnectorInfoError, 0, len(c.Status.Tasks))
+
 		runningTasks := 0
+		failedTasks := 0
+		pausedTasks := 0
+		restartingTasks := 0
 		for i, task := range c.Status.Tasks {
 			tasks[i] = ClusterConnectorTaskInfo{
 				TaskID:   task.ID,
 				State:    task.State,
 				WorkerID: task.WorkerID,
 				Trace:    task.Trace,
 			}
-			if task.State == connectorStateRunning {
+
+			switch task.State {
+			case connectorStateRunning:
 				runningTasks++
+			case connectorStateFailed:
+				failedTasks++
+
+				errTitle := fmt.Sprintf("Connector %s Task %d is in failed state", c.Info.Name, task.ID)
+				connectorTaskErrors = append(connectorTaskErrors, ClusterConnectorInfoError{
+					Type:    connectorErrorTypeError,
+					Title:   errTitle,
+					Content: errorContentFromTrace(errTitle, task.Trace),
+				})
+			case connectorStatePaused:
+				pausedTasks++
+			case connectorStateRestarting:
+				restartingTasks++
 			}
 		}
 
+		// See const definitions for rules
+		var connStatus connectorStatus
+		//nolint:gocritic // this if else is easier to read as they map to rules specified above
+		if (c.Status.Connector.State == connectorStateRunning) && (runningTasks == len(c.Status.Tasks)) {
+			connStatus = connectorStatusHealthy
+		} else if c.Status.Connector.State == connectorStateFailed {
+			connStatus = connectorStatusUnhealthy
+		} else if (c.Status.Connector.State == connectorStateRunning) && (len(c.Status.Tasks) == 0 || (failedTasks > 0)) {
+			connStatus = connectorStatusDegraded
+		} else if (c.Status.Connector.State == connectorStatePaused) || (len(c.Status.Tasks) > 0 && (pausedTasks > 0)) {
+			connStatus = connectorStatusPaused
+		} else if (c.Status.Connector.State == connectorStateRestarting) || (len(c.Status.Tasks) > 0 && (restartingTasks > 0)) {
+			connStatus = connectorStatusRestarting
+		}
+
+		connectorErrors := make([]ClusterConnectorInfoError, 0)
+		if c.Status.Connector.State == connectorStateFailed {
+			errTitle := "connector " + c.Info.Name + " is in failed state"
+			connectorErrors = append(connectorErrors, ClusterConnectorInfoError{
+				Type:    connectorErrorTypeError,
+				Title:   errTitle,
+				Content: errorContentFromTrace(errTitle, c.Status.Connector.Trace),
+			})
+		} else if len(c.Status.Connector.Trace) > 0 {
+			errTitle := "connector " + c.Info.Name + " has an error"
+			connectorErrors = append(connectorErrors, ClusterConnectorInfoError{
+				Type:    connectorErrorTypeError,
+				Title:   errTitle,
+				Content: errorContentFromTrace(errTitle, c.Status.Connector.Trace),
+			})
+		}
```

Comment:
I think we also want to inject our own error in case the connector spawned zero tasks. There are some connectors which report a ready state but don't spawn any connectors in case there's a configuration issue. @tomasz-sadura can probably tell how to reproduce this behaviour.

Reply:
yes, one example is to use MirrorSourceConnector for non-existing topic, e.g.
```
{
    "connector.class": "org.apache.kafka.connect.mirror.MirrorSourceConnector",
    "topics": "non-existing-topic",
    "source.cluster.bootstrap.servers": "redpanda:29092",
    "source.cluster.security.protocol": "PLAINTEXT",
    "source.cluster.alias": "source",
    "replication.factor": "-1",
    "offset-syncs.topic.replication.factor": "-1",
    "name": "mirror-source-connector"
}
```

Reply:
`errTitle := "connector " + c.Info.Name + " has an error"`

first letter should be uppercase in all titles